grype/match/matches_test.go

```diff
@@ -0,0 +1,84 @@
+package match
+
+import (
+	"testing"
+
+	"github.com/anchore/grype/grype/pkg"
+	"github.com/anchore/grype/grype/vulnerability"
+	syftPkg "github.com/anchore/syft/syft/pkg"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMatchesSort(t *testing.T) {
```

Comment:
It isn't clear to me what cases this is covering. While trying to ensure there was coverage for the sorting, I had to go back and forth to catch which ones where for the same ID, then for the same package name and so on. This is not only time-consuming, but if I had to fix (or update) the sorting, I wouldn't be sure how to add a specific case that covers the new change. 

Going through the matches, it seems there is no case that catches same ID, same package name, and same Version.

I'd suggest breaking the test cases up so that the conditions they are testing are explicit.

Reply:
I kept the original test for a multi-dimensional sort test, but also broke out testing into pairs within the same dimension. 👍